### ISSUE TITLE
Question regarding logging additional information with tensorboard?

### ISSUE THREAD

**Issue by uduse** [guildai/guildai#290]
Hi there, I would like to first thank the developers for this awesome project! 🤗 🤗  I've been enjoying using it.

I'm thinking about logging additional information to tensorboard (e.g., [histogram](https://github.com/tensorflow/tensorboard/blob/master/docs/r1/histograms.md) ) during a run, but I'm not sure how to associate those logs with the existing logs. A solution could be logging that information to the same directory as the `output-scalars` directory. However, it is unclear to me how to fetch the target log directory at runtime. Does guild.ai set some kind of environment variable for this purpose?

**Comment by uduse** [comment]
I found the environment variable `RUN_DIR` by print out all environment variables of a run. Somehow I wasn't able to find this piece of information in the docs 😢  Is this the right thing to use?

**Comment by samedii** [comment]
What is not working when you use tensorboard normally? It is supposed to "just work" with `guild tensorboard XYZ`

**Comment by uduse** [comment]
@samedii I didn't know about `RUN_DIR` so I wasn't sure how should I set my `logdir` for writing tensorboard events. I played with it a bit and found that using `RUN_DIR` perfectly solves the problem 😄

**Comment by samedii** [comment]
You shouldn't need `RUN_DIR`. You can just put the tensorboard logs in something like "./tb" since "." will point to the `RUN_DIR`

E.g. I use tensorboard with guild + pytorch like this:
```python
tensorboard_logger = torch.utils.tensorboard.SummaryWriter(log_dir="tb")
```

```bash
guild tensorboard 1
```

**Comment by uduse** [comment]
@samedii Oh, so this means as long as I write to a place under the current directory (implicitly relative to `RUN_DIR`), it will be gathered by guild 🤔  This makes sense! Thanks a lot! 🙇‍♂️

**Comment by gar1t** [comment]
There are a number of environment variables that Guild sets, but you're right they're not documented! I've opened an internal issue to fix that.

But @samedii is spot on - you don't need to reference this directory as Guild ensures that the current working directory is set to the run directory when your code runs. This lets you write anything you want as a relative path and it will appear as a run artifact.

This includes any TF event files. The only consideration is the relative path, which will be reflected in TensorBoard whenever relative paths are shown. E.g. a common pattern is to write training stats under `./train` and test/eval stats under `./eval` using the same tag (e.g. `accuracy`) this keeps them differentiated in TensorBoard.

If you're logging histograms, you might also consider using TF event logs for everything else, including scalars. This is more efficient than relying on Guild to monitor script output for "output scalar" patterns. In this case you log explicitly and disable Guild's output scalar feature this way:

``` yaml
op:
  output-scalars: off
```

Guild supports output scalars because not everyone wants to use TF event logs. But since you're doing that already with histograms, you might consider logging scalars that way as well.

**Comment by uduse** [comment]
@gar1t Thanks for the explanation!

> If you're logging histograms, you might also consider using TF event logs for everything else, including scalars.

Yeah I also realized this is the better thing to do, so I don't have to mess around with regex parsing 😵
